vcs/src/test/java/org/openjdk/skara/vcs/openjdk/converter/GitToHgConverterTests.java

```diff
@@ -400,16 +400,121 @@ void convertMergeCommitWithP1Diff() throws IOException {
+
+    private void cloneAndConvertAndVerify(String repo) throws IOException {
+        try (var hgRoot = new TemporaryDirectory(false);
+             var gitRoot = new TemporaryDirectory(false)) {
+            var gitRepo = Repository.clone(URI.create("https://git.openjdk.java.net/" + repo + ".git"), gitRoot.path());
+            var hgRepo = Repository.init(hgRoot.path(), VCS.HG);
+            var converter = new GitToHgConverter(new Branch("master"));
+            var marks = converter.convert(gitRepo, hgRepo);
+            assertReposEquals(marks, gitRepo, hgRepo);
         }
     }
+
+    @Test
+    void convertGitTag() throws IOException {
+        try (var hgRoot = new TemporaryDirectory(false);
+             var gitRoot = new TemporaryDirectory(false)) {
+            var gitRepo = Repository.init(gitRoot.path(), VCS.GIT);
+            var readme = gitRoot.path().resolve("README.md");
+
+            Files.writeString(readme, "First line\n");
+            gitRepo.add(readme);
+            gitRepo.commit("First line", "Foo Bar", "foo@openjdk.java.net");
+
+            Files.writeString(readme, "Second line\n", StandardOpenOption.APPEND);
+            gitRepo.add(readme);
+            var second = gitRepo.commit("Second line", "Foo Bar", "foo@openjdk.java.net");
+            var tagDate = ZonedDateTime.parse("2020-08-24T11:30:32+02:00");
+            var tag = gitRepo.tag(second, "1.0", "Added tag 1.0", "Foo Bar", "foo@openjdk.java.net", tagDate);
+
+            var hgRepo = Repository.init(hgRoot.path(), VCS.HG);
+            var converter = new GitToHgConverter();
+            var marks = converter.convert(gitRepo, hgRepo);
+            var lastMark = marks.get(marks.size() - 1);
+            assertEquals(second, lastMark.git());
+            assertTrue(lastMark.tag().isPresent());
+
+            Files.writeString(readme, "Third line\n");
+            gitRepo.add(readme);
+            gitRepo.commit("Third line", "Foo Bar", "foo@openjdk.java.net");
+
+            converter = new GitToHgConverter();
+            var newMarks = converter.convert(gitRepo, hgRepo, marks);
+            var hgCommits = hgRepo.commitMetadata(true);
+            assertEquals(4, hgCommits.size());
+            assertEquals(List.of("First line"), hgCommits.get(0).message());
+            assertEquals(List.of("Second line"), hgCommits.get(1).message());
+            assertEquals(List.of("Added tag 1.0"), hgCommits.get(2).message());
+            assertEquals(List.of("Third line"), hgCommits.get(3).message());
+            assertEquals(List.of(new Tag("tip"), new Tag("1.0")), hgRepo.tags());
+
+            var annotated = hgRepo.annotate(new Tag("1.0"));
+            assertTrue(annotated.isPresent());
+            assertEquals("foo", annotated.get().author().name());
+            assertEquals(tagDate, annotated.get().date());
+            assertEquals("Added tag 1.0", annotated.get().message());
+        }
+    }
+
+    @Disabled("Depends on internet connection")
```

Comment:
There are similar tests for the bridge bot that are skipped if there is no connection:

https://github.com/openjdk/skara/blob/269027247e5c5778560d3031476dfd0ece4e57d3/bots/hgbridge/src/test/java/org/openjdk/skara/bots/hgbridge/BridgeBotTests.java#L131

but not sure if something similar would be appropriate here as well?

Reply:
Good idea, but these takes also takes a bit too long to run in addition to requiring an internet connection, so I would prefer to leave them out for now. I might look into conditionally enable them depending on an environment variable so that we can have them running in CI.